tests/neg/inline-param-unstable-path.scala

```diff
@@ -0,0 +1,6 @@
+inline val a = 3
+inline def f(inline x: Int, y: Int, z: => Int): Unit =
+  val x2: x.type = x // error: (x : Int) is not a valid singleton type, since it is not an immutable path
```

Comment:
Does this also handle:
```suggestion
  inline val y = x
  val y2: y.type = y
```

Reply:
It is checked after the code is inlined. I will add a test.